pkg/cmd/component/cmd/ui/ui.go

```diff
@@ -403,3 +403,81 @@ func AddEnvironmentName() string {
 	handleError(err)
 	return strings.TrimSpace(environmentName)
 }
+func ApplicationOutputPath() string {
+	var path string
+	prompt := &survey.Input{
+		Message: "Provide a path to application folder",
+		Help:    "This is the path where application configuration is stored locally before you push it to the repository GitRepoURL",
+	}
+	err := survey.AskOne(prompt, &path, validateAppPath())
+	handleError(err)
+	return strings.TrimSpace(path)
+}
+
+func validateAppPath() survey.Validator {
+	return func(input interface{}) error {
+		return validateAppFolder(input)
+	}
+}
+
+func validateAppFolder(input interface{}) error {
+	if u, ok := input.(string); ok {
+		exists, _ := ioutils.IsExisting(ioutils.NewFilesystem(), u)
+		if !exists {
+			appName := strings.Split(u, "/")
+			err := ValidateName(appName[len(appName)-1])
+			if err != nil {
+				err := fmt.Errorf(err.Error())
+				return (err)
+			}
+
+		} else {
+			l := ListFiles(u)
+			if len(l) != 0 {
+				exists, _ = ioutils.IsExisting(ioutils.NewFilesystem(), filepath.Join(u, "components"))
```

Comment:
This function probably could have a testcase written for it.  Recall here that the in memory filesystem is used instead.

Reply:
There is no test case written for ui.go

Reply:
This function doesn't have prompts and only validates. It could be moved out.